DataFormats/Headers/include/Headers/Stack.h

```diff
@@ -56,11 +56,13 @@ struct Stack {
   value_type* data() const { return buffer.get(); }
   size_t size() const { return bufferSize; }
   allocator_type get_allocator() const { return allocator; }
+  const BaseHeader* first() const {return reinterpret_cast<const BaseHeader*>(this->data());}
```

Comment:
why not implement `first()` using `firstHeader()`?

Reply:
member function first() is fast, does no checking and just returns a pointer to a freshly constructed **valid** stack (since ctor makes sure it is valid).
the static helper firstHeader() makes no assumptions about what it gets (raw buffer pointer) and goes through the motions of trying to guess if the contents actually look like a header... so many choices... :)